spec.bs

```diff
@@ -2179,27 +2216,45 @@ Width and height of an ad.
+<pre class="idl">
+dictionary ScoreAdOutput {
+  required double desirability;
+  double bid;
+  boolean allowComponentAuction = false;
+};
+</pre>
+Either a dictionary of this type, or a {{double}}, are handled as the return values.
 
-<dl dfn-for="score ad output">
-: <dfn>desirability</dfn>
-:: A {{double}}.
-  Numeric score of the bid. Must be positive or the ad will be rejected. The winner of the auction
+The meanings of the fields are as follows:
+<dl class=domintro>
+  <dt>{{ScoreAdOutput/desirability}}
+  <dd>Numeric score of the bid. Must be positive or the ad will be rejected. The winner of the auction
   is the bid which was given the highest score.
-: <dfn>allow component auction</dfn>
-:: A [=boolean=].
-   If the bid being scored is from a component auction and this value is not true, the bid is
-   ignored. This field must be present and true both when the component seller scores a bid, and
-   when that bid is being scored by the top-level auction.
-: <dfn>bid</dfn>
-:: Null or a {{double}}.
-  Is null if the auction has no component auction, or if the auction is a top-level auction.
-  Modified bid value to provide to the top-level seller script. If present, this will be passed to
-  the top-level seller's `scoreAd()` and `reportResult()` methods instead of the original bid, if
-  the ad wins the component auction and top-level auction, respectively.
-
+  <dt>{{ScoreAdOutput/bid}}
+  <dd>Only relevant if this is a component auction.  If present, this will be passed to the top-level seller's `scoreAd()` and `reportResult()` methods instead of the original bid, if the ad wins the component auction and top-level auction, respectively.
+  <dt>{{ScoreAdOutput/allowComponentAuction}}
+  <dd>If the bid being scored is from a component auction and this value is not true, the bid is ignored. This field must be present and true both when the component seller scores a bid, and when that bid is being scored by the top-level auction.
 </dl>
 
+TODO: This also has an ad field, which should behave similar to the way {{ScoreAdOutput/bid}} affects [=generated bid/modified bid=], and then affecting the adMetadata parameter to scoreAd.
+
+<div algorithm>
+
+To <dfn>process scoreAd output</dfn> given an [=ECMAScript/Completion Record=] |result|:
+  1. If |result| is an an [=abrupt completion=], return failure.
+  1. If |result|.\[[Value]] is a [=Number=]:
+    1. Let |checkedScore| be the result of [=converted to an IDL value|converting=] |result|.\[[Value]] to a {{double}}.
+    1. If an exception was [=thrown=] in the previous step, return failure.
+    1. Let |resultIDL| be a new {{ScoreAdOutput}}.
+    1. Set |resultIDL|'s {{ScoreAdOutput/desirability}} to |checkedScore|.
+    1. Return |resultIDL|.
+  1. Let |resultIDL| be the result of [=converted to an IDL value|converting=]
+          |result|.\[[Value]] to a {{ScoreAdOutput}}.
+  1. If an exception was [=thrown=] in the previous step, return failure.
+  1. Return |resultIDL|

```

Comment:
nit: period at the end of this sentence.

Reply:
Done